crates/exex/src/exex.rs

```diff
@@ -148,10 +153,16 @@ impl<Node: FullNodeComponents> KakarotRollup<Node> {
         // Process each block and its transactions.
         for (block, txs, _) in blocks {
             // Execute the block and handle the result.
-            match execute_block(&mut self.db, block, txs).await {
+            match execute_block(&mut self.db, block, txs.clone()).await {
                 Ok((block, bundle, _, _)) => {
-                    // Seal the block and insert it into the database.
+                    // Seal the block.
                     let block = block.seal_slow();
+                    // After the block is executed, we want to run the get_env program
+                    //
+                    // The goal is to store in a Cairo program, via hint, all the information about
+                    // the block
+                    self.execute_get_env(block.block.clone(), txs, chain.tip().number)?;
```

Comment:
Here imo this is the best place to put this execution at the moment to be able to catch both block and transactions.

Reply:
let's keep this like that for this PR if you prefer. I'll update the struct afterwards with the kakarot fork, then we'll update based on the real struct

Reply:
Sure let us do like this